tc/tc_util.h

```diff
@@ -24,14 +24,14 @@ struct qdisc_util {
 	struct  qdisc_util *next;
 	const char *id;
 	int (*parse_qopt)(struct qdisc_util *qu, int argc,
-			  char **argv, struct nlmsghdr *n);
+			  char **argv, struct nlmsghdr *n, char *dev);
 	int (*print_qopt)(struct qdisc_util *qu,
 			  FILE *f, struct rtattr *opt);
 	int (*print_xstats)(struct qdisc_util *qu,
 			    FILE *f, struct rtattr *xstats);
 
 	int (*parse_copt)(struct qdisc_util *qu, int argc,
-			  char **argv, struct nlmsghdr *n);
+			  char **argv, struct nlmsghdr *n, char *dev);
```

Comment:
Hi Cody, after looking through the codebase, `struct qdisc_util` is written strictly for general handling of the qdisc (and not just specifically how the qdisc is being used, by the tc command), so much so that even the netlink message being sent consists purely of local variables defined in the specific qdisc file (which is invoked by `parse_qopt`). 

The problem with adding `dev` as a member definition of `qdisc_util` is that, in the codebase, the other members of `qdisc_util` (which of relevance here is `id`) are being used in such a way that the only function they serve is to identify which particular qdisc is being invoked by looking at the arguments in the command.  (And then, that qdisc file is invoked through `parse_opt` where all the values being passed to the kernel are initialised, as local variables)

Given this, using a member definition of `qdisc_util` (i.e the `dev` parameter) which is used to describe how the qdisc is being applied, seems to make things difficult and complicated. From looking at how the `qdisc_util` code is being handled in the codebase, I do not think a member `qdisc_util` is meant to be used in the way we intend to use it.